packages/integration-tests/src/getProperties.test.ts

```diff
@@ -3,42 +3,28 @@ import { getProperties } from "joist-orm";
 
 describe("getProperties", () => {
   it("should work", () => {
-    expect(getProperties(bookMeta)).toMatchInlineSnapshot(`
-      Array [
-        "advances",
-        "reviews",
-        "author",
-        "image",
-        "tags",
-      ]
-    `);
+    expect(getProperties(bookMeta)).toEqual(expect.arrayContaining(["advances", "reviews", "author", "image", "tags"]));
```

Comment:
Do we need `arrayContaining`? 

Fwiw I slightly prefer the inline snapshots b/c they are so ridiculously easy to update.

I.e. when doing the refactoring for tagged ids in graphql-service, the tests that had ids in inline snapshots were basically insta-fixed + just post-review the diff to verify they make sense, vs. the ones that had ids in random strings in `.toEqual`'s that I had to hunt and peck update. I probably spent ~2-3 hours updating tests that could have been could have been insta-fixed (if in the extreme scenario which I'm not seriously pitching) if they used snapshots.

So not saying I'd use `toMatchInlineSnapshot` for every assertion ever (...thinking about it...), but just communicating that I specifically like how "agile" they are/pleasant they make updating test assertions while refactoring.

Just musing, but if Jest had a way to auto-update failures to `.toEqual` calls that were constants, that would be pretty slick.

Reply:
For this specific line, no we don't need it.  Just `toEqual` is sufficient.  For the other 5 assertions in the file, we need it since we are testing for specific things and shouldn't need to be exhaustive.

I don't really get the purpose of `toMatchInlineSnapshot`.  It seems like added boilerplate without any syntax checking.  All its use cases are handled by a combination of `toEqual`, `toMatchObject` and `expect.arrayContaining`.

Reply:
> shouldn't need to be exhaustive

Ah sure, I didn't mind it for these, although snapshots asserting unrelated things is generally undesirable, especially when you get into external snapshots like `toMatchSnapshot`.

> I don't really get the purpose of toMatchInlineSnapshot.

The purpose is how damn fast it is b/c you don't type anything. You know I strongly believe in static typing, but it's really common for assertions to be "does this `result` DTO/POJO match-ish what I expect" is the singular intent and having to "encode" that intent with ".toEqual ... type a bunch of syntax chars" or "arrayContaining ... type a bunch of syntax chars" doesn't actually add any value to achieving the singular intent of "does this DTO/tree match-ish what I expect". It's just extra typing to placate "how we've always done things".

(I say this as still a very strong `toMatchSnapshot` skeptic and an initial `toMatchInlineSnapshot` skeptic, but I've since come around on the inline version. You get used to "sure it's a string with some extra backets" and weigh the "small con" of "sure this is not syntax highlighted" with the imo "large pro" of "wow I didn't have to type any of this and can now insta-fix assertions for updated tests" and, eh, I've personally found it useful. At least to the extent that I think wholesale removing it from codebases in unrelated PRs is unnecessary.)

(I also very rarely use "less keystrokes" as an argument given vim bindings/"you're a programmer so get good at typing"/etc. But despite "yes I'm already fast" I find judicious `toMatchineInlineSnapshot` makes me (a tiny bit) faster.)

(Also, I'll repeat the idea that if you want to fork Jest and build a `.toEqual`s that will similarly auto-update assertions that are all consts of hashes/lists, and so achieve the same workflow productivity without the "yuck this is a string", I personally think that'd be pretty cool.)

Reply:
You don't need to type anything with `toEqual` either. You can get the actual result as a json string, at least you can in webstorm.

For `toMatchObject` or `arrayContains` you're only looking for a subset of the result anyways and imo that should always be explicitly defined so that intent can be understood.

Reply:
>  You can get the actual result as a json string, at least you can in webstorm and then it's just a copy/paste.

?

Reply:
> you're only looking for a subset of the result anyways

Definitely agreed as a rule, with the disclaimer I think it's like DRY, where 100% adherence is not required / judicious exceptions is fine.

A subset version (i.e. `toMatchObject`-ish semantics) of `toMatchInlineSnapshot` would be great. :-P :-) (Agreed `arrayContains` is too specific to try and auto-update on first-assertion/updating-assertions.)

Reply:
If a `toEquals` test fails for me, Webstorm has a '<click to see difference>' link that shows you what the actual result was as json string.  If you really want to be as quick as possible, you can just copy it, then paste it over the previous expectation.  

IMO, some friction when updating tests is actually a good thing.  If it's too easy then it encourages not actually validating the results.